<h1> Palmer Penguin Data Analytics </h1>
<hr> 
<h3>Analyst: Jeanne Nicole Gianan</h3>

In [3]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<p>Importing dataset to Jupyter notebook</p>
<p>Converting dataset to dataframe</p>
<ul>Requirements
    <li>Name of the dataframe</li>
    <li>Source</li>
</ul>

In [2]:
penguins = pd.read_csv('datasets/penguins.csv')

In [68]:
#Incase of error, dahil sa CoCalc ko po sya ginawa.
#penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3>Descriptive Analytics</h3>

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3>Cleaning the Data Frame</h3>

In [6]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h2>Selecting a column</h2>
<h3>Syntax</h3><br>
data_frame['column_name']

In [10]:
penguins['island']

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
339        Dream
340        Dream
341        Dream
342        Dream
343        Dream
Name: island, Length: 344, dtype: object

In [11]:
penguins['island'] == 'Dream'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: island, Length: 344, dtype: bool

In [12]:
penguins[penguins['island'] == 'Dream']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

In [13]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [14]:
adelie_df = penguins[penguins['species'] == "Adelie"]

In [15]:
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [16]:
gentoo_df = penguins[penguins['species'] == "Gentoo"]

In [17]:
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [18]:
chinstrap_df = penguins[penguins['species'] == "Chinstrap"]

In [19]:
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [20]:
penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [21]:
penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


In [22]:
penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000) & (penguins['sex'] == 'female')].info()

<class 'pandas.core.frame.DataFrame'>
Index: 1 entries, 284 to 284
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            1 non-null      object 
 1   island             1 non-null      object 
 2   bill_length_mm     1 non-null      float64
 3   bill_depth_mm      1 non-null      float64
 4   flipper_length_mm  1 non-null      float64
 5   body_mass_g        1 non-null      float64
 6   sex                1 non-null      object 
 7   year               1 non-null      int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 72.0+ bytes


In [23]:
penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000) & (penguins['sex'] == 'male')].info()

<class 'pandas.core.frame.DataFrame'>
Index: 15 entries, 289 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            15 non-null     object 
 1   island             15 non-null     object 
 2   bill_length_mm     15 non-null     float64
 3   bill_depth_mm      15 non-null     float64
 4   flipper_length_mm  15 non-null     float64
 5   body_mass_g        15 non-null     float64
 6   sex                15 non-null     object 
 7   year               15 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


<h1>HOW TO FILL UP NULL VALUE</h1>

In [24]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [25]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [26]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [27]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [28]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [29]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [30]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [31]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [32]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [33]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h1>HOW TO CREATE A COLUMN</h1>

In [34]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [35]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [36]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000, 2)

In [37]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000           18.7         181.000000   
1       Adelie  Torgersen        39.50000           17.4         186.000000   
2       Adelie  Torgersen        40.30000           18.0         195.000000   
3       Adelie  Torgersen        43.92193            NaN         200.915205   
4       Adelie  Torgersen        36.70000           19.3         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000           19.8         207.000000   
340  Chinstrap      Dream        43.50000           18.1         202.000000   
341  Chinstrap      Dream        49.60000           18.2         193.000000   
342  Chinstrap      Dream        50.80000           19.0         210.000000   
343  Chinstrap      Dream        50.20000           18.7         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007     3.75  
1    3800.000000  female  2007     3.80  
2    3250.000000  female  2007     3.25  
3    4201.754386    male  2007     4.20  
4    3450.000000  female  2007     3.45  
..           ...     ...   ...      ...  
339  4000.000000    male  2009     4.00  
340  3400.000000  female  2009     3.40  
341  3775.000000    male  2009     3.78  
342  4100.000000    male  2009     4.10  
343  3775.000000  female  2009     3.78  

[344 rows x 9 columns]

<h1>EXPLORATORY DATA ANALYTICS (EDA)</h1>

<h3>1. Which species lives the most in the dream island?</h3>

In [38]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [39]:
dream = penguins[penguins['island'] == "Dream"]

In [40]:
dream

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
30        3250.0  female  2007     3.25  
31        3900.0    male  2007     3.90  
32        3300.0  female  2007     3.30  
33        3900.0    male  2007     3.90  
34        3325.0  female  2007     3.32  
..           ...     ...   ...      ...  
339       4000.0    male  2009     4.00  
340       3400.0  female  2009     3.40  
341       3775.0    male  2009     3.78  
342       4100.0    male  2009     4.10  
343       3775.0  female  2009     3.78  

[124 rows x 9 columns]

In [41]:
dream['species'].value_counts()

species
Chinstrap    68
Adelie       56
Name: count, dtype: int64

<h3>Insight #1 </h3>
<h4>The species that lives the most in the dream island is the Chinstrap species with the number of 68, compared to the Adelie species which only has 56. We can also see that there are no Gentoo species living in the dream island</h4>

<h3>2. In the year 2008, how many species are there? </h3>

In [42]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000           18.7         181.000000   
1       Adelie  Torgersen        39.50000           17.4         186.000000   
2       Adelie  Torgersen        40.30000           18.0         195.000000   
3       Adelie  Torgersen        43.92193            NaN         200.915205   
4       Adelie  Torgersen        36.70000           19.3         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000           19.8         207.000000   
340  Chinstrap      Dream        43.50000           18.1         202.000000   
341  Chinstrap      Dream        49.60000           18.2         193.000000   
342  Chinstrap      Dream        50.80000           19.0         210.000000   
343  Chinstrap      Dream        50.20000           18.7         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007     3.75  
1    3800.000000  female  2007     3.80  
2    3250.000000  female  2007     3.25  
3    4201.754386    male  2007     4.20  
4    3450.000000  female  2007     3.45  
..           ...     ...   ...      ...  
339  4000.000000    male  2009     4.00  
340  3400.000000  female  2009     3.40  
341  3775.000000    male  2009     3.78  
342  4100.000000    male  2009     4.10  
343  3775.000000  female  2009     3.78  

[344 rows x 9 columns]

In [43]:
eight = penguins[penguins['year'] == 2008]

In [44]:
eight

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
50      Adelie  Biscoe            39.6           17.7              186.0   
51      Adelie  Biscoe            40.1           18.9              188.0   
52      Adelie  Biscoe            35.0           17.9              190.0   
53      Adelie  Biscoe            42.0           19.5              200.0   
54      Adelie  Biscoe            34.5           18.1              187.0   
..         ...     ...             ...            ...                ...   
315  Chinstrap   Dream            53.5           19.9              205.0   
316  Chinstrap   Dream            49.0           19.5              210.0   
317  Chinstrap   Dream            46.2           17.5              187.0   
318  Chinstrap   Dream            50.9           19.1              196.0   
319  Chinstrap   Dream            45.5           17.0              196.0   

     body_mass_g     sex  year  mass_kg  
50        3500.0  female  2008     3.50  
51        4300.0    male  2008     4.30  
52        3450.0  female  2008     3.45  
53        4050.0    male  2008     4.05  
54        2900.0  female  2008     2.90  
..           ...     ...   ...      ...  
315       4500.0    male  2008     4.50  
316       3950.0    male  2008     3.95  
317       3650.0  female  2008     3.65  
318       3550.0    male  2008     3.55  
319       3500.0  female  2008     3.50  

[114 rows x 9 columns]

In [45]:
eight['species'].value_counts()

species
Adelie       50
Gentoo       46
Chinstrap    18
Name: count, dtype: int64

<h3>Insight #2</h3>
<h4>As we can see there are a total of 114 species in the year 2008. Specifically, Adelie with a no. of 50, Gentoo with the no. od 46 and lastly Chinstrap with a no. 18, with the total no. 114</h4>

<h3>3. What is the most common bill length in mm?</h3>

In [46]:
penguins['bill_length_mm'].mode()

0    41.1
Name: bill_length_mm, dtype: float64

<h3>Insight #3</h3>
<h4>The most common bill length in millimeters is 41.1</h4>

<h3>4. What is the average body mass of penguin on Biscoe Island?</h3>

In [47]:
biscoe = penguins[penguins['island'] == "Biscoe"]

In [48]:
biscoe['body_mass_g'].mean()

np.float64(4712.95687134503)

<h3>Insight #4</h3>
<h4>The average body mass of penguins living in Biscoe Island is 4712.96</h4>

<h3>5. How many penguins are male and how many are female</h3>

In [49]:
penguins['sex'].value_counts()

sex
male      179
female    165
Name: count, dtype: int64

<h3>Insight #5</h3>
<h4>The no. of male are 179 and female are 165. We can see that there are more male than female</h4>

<h3>6. How many species belong to each species?</h3>

In [50]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h3>Insight #6</h3>
<h4>There are 152 Adelie, 124 Gentoo and 68 Chinstrap. A total no. of 344 species. There are more Adelie species compared to the other 2 species.</h4>

<h3>7. List all the unique island name where penguins were obsrved and which island hosts the most penguins?</h3>

In [51]:
penguins['island'].unique()

array(['Torgersen', 'Biscoe', 'Dream'], dtype=object)

In [52]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h3>Insight #7</h3>
<h4>There are a 3 island namely Torgersen, Biscoe and Dream. The Biscoe island host the most penguin with a total no. of 168</h4>

<h1>DATA AGGREGATION</h1>

ilan species per island ang nakatira?

In [53]:
PerIsland = penguins.groupby(['species', 'island']).count()

In [54]:
PerIsland

bill_length_mm  bill_depth_mm  flipper_length_mm  \
species   island                                                        
Adelie    Biscoe                 44             44                 44   
          Dream                  56             56                 56   
          Torgersen              52             51                 52   
Chinstrap Dream                  68             68                 68   
Gentoo    Biscoe                124            123                124   

                     body_mass_g  sex  year  mass_kg  
species   island                                      
Adelie    Biscoe              44   44    44       44  
          Dream               56   56    56       56  
          Torgersen           52   52    52       52  
Chinstrap Dream               68   68    68       68  
Gentoo    Biscoe             124  124   124      124

In [55]:
avg_species_island = penguins.groupby(['species', 'island'])['body_mass_g'].mean()

In [56]:
avg_species_island

species    island   
Adelie     Biscoe       3709.659091
           Dream        3688.392857
           Torgersen    3715.899123
Chinstrap  Dream        3733.088235
Gentoo     Biscoe       5068.965761
Name: body_mass_g, dtype: float64

<h1>Data Aggregation</h1>
<h2> The use of .groupby()</h2>

In [57]:
penguins_insight8 = penguins.groupby(['species', 'island', 'sex', 'year',]).size()

In [58]:
penguins_insight8

species    island     sex     year
Adelie     Biscoe     female  2007     5
                              2008     9
                              2009     8
                      male    2007     5
                              2008     9
                              2009     8
           Dream      female  2007     9
                              2008     8
                              2009    10
                      male    2007    11
                              2008     8
                              2009    10
           Torgersen  female  2007     8
                              2008     8
                              2009     8
                      male    2007    12
                              2008     8
                              2009     8
Chinstrap  Dream      female  2007    13
                              2008     9
                              2009    12
                      male    2007    13
                              2008     9
                      

<h1>TESTING CORRELATION</h1> 

<h3>Pearson R Correlation</h3>
<hr>
<p>Used for testing correlation between two <strong>NUMERICAL VALUES</strong></p>

In [59]:
weight_flipper_cor = penguins['body_mass_g'].corr(penguins['flipper_length_mm'])

In [60]:
weight_flipper_cor

np.float64(0.8712017673060115)

<h3>8. What is the total body mass of species living in Torgersen</h3>



In [61]:
is_dream = penguins[penguins['island'] == 'Dream']

In [62]:
is_dream

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
30        3250.0  female  2007     3.25  
31        3900.0    male  2007     3.90  
32        3300.0  female  2007     3.30  
33        3900.0    male  2007     3.90  
34        3325.0  female  2007     3.32  
..           ...     ...   ...      ...  
339       4000.0    male  2009     4.00  
340       3400.0  female  2009     3.40  
341       3775.0    male  2009     3.78  
342       4100.0    male  2009     4.10  
343       3775.0  female  2009     3.78  

[124 rows x 9 columns]

In [63]:
total_body_mass = is_dream.groupby('species')['body_mass_g'].sum()

In [64]:
total_body_mass

species
Adelie       206550.0
Chinstrap    253850.0
Name: body_mass_g, dtype: float64

<h3>Insight #8</h3>
<h4>There are 2 species in Dream Island, the Adelei and Chinstrap. We can see that Chinstrap is heavier than Adelei with a total body mass of 253850.</h4>

<h3>9. What is the median bill depth of penguins for each species and sex?</h3>

In [65]:
penguins.groupby(['species', 'sex'])['bill_depth_mm'].median()

species    sex   
Adelie     female    17.60
           male      18.90
Chinstrap  female    17.65
           male      19.30
Gentoo     female    14.25
           male      15.70
Name: bill_depth_mm, dtype: float64

<h3>Insight #9</h3>
<h4>We will observe that the female species has smaller bill depth in median compared to male, regardless of their species</h4>

<h3>10. For each year, what is the heaviest body mass recorded?</h3>

In [7]:
penguins.groupby('year')['body_mass_g'].max()

year
2007    6300.0
2008    6000.0
2009    6000.0
Name: body_mass_g, dtype: float64

<h3>Insight #10</h3>
<h4>The year 2007 has recorded the heaviest body mass recorded with a no. 6300g of 6.3kg, while 2008 and 2009 are the same.</h4>

<h3>11.How many species are recorded for each year?</h3>

In [67]:
penguins.groupby('year')['species'].size()

year
2007    110
2008    114
2009    120
Name: species, dtype: int64

<h3>Insight #11</h3>
<h4>We can see that the no. of species recorded each year is ascending with a total of 120 species recorded in 2009.</h4>

<h3>12. For each species, what is the correlation between bill length and bill depth?</h3>

In [9]:
penguins.groupby('species')[['bill_length_mm', 'bill_depth_mm']].corr()

bill_length_mm  bill_depth_mm
species                                                
Adelie    bill_length_mm        1.000000       0.391492
          bill_depth_mm         0.391492       1.000000
Chinstrap bill_length_mm        1.000000       0.653536
          bill_depth_mm         0.653536       1.000000
Gentoo    bill_length_mm        1.000000       0.643384
          bill_depth_mm         0.643384       1.000000

<h3>Insight #12</h3>
<h4>When it come to Adelei species, the bill length does not have any correlation with the bill depth, but from the remaining 2 species the bill length does have a correlation with the bill depth.</h4>